app/scripts/lib/setupSentry.js

```diff
@@ -118,16 +118,20 @@ export default function setupSentry({ release, getState }) {
    * @returns `true` if MetaMask's state has been initialized, and MetaMetrics
    * is enabled, `false` otherwise.
    */
-  function getMetaMetricsEnabled() {
-    if (getState) {
-      const appState = getState();
-      if (!appState?.store?.metamask?.participateInMetaMetrics) {
-        return false;
-      }
-    } else {
+  async function getMetaMetricsEnabled() {
```

Comment:
Later in this file this function is called synchronously in startSession/endSession. Those methods should be able to be made async without any complications.

Reply:
Good catch. I forgot to re-review this after rebasing, which brought in your session changes.

Reply:
https://github.com/MetaMask/metamask-extension/issues/20299 I also just added this

Reply:
Fixed here: https://github.com/MetaMask/metamask-extension/pull/20265/commits/c27e1de4de813bda0ec0187772d6fc8379d38796